cupy/random/_generator_api.pyx

```diff
@@ -219,6 +222,49 @@ class Generator:
         # omitted args.
         return (<object>y).astype(dtype, copy=False)
 
+    def gamma(self, shape, scale=1.0, size=None):
+        """Returns an array of samples drawn from a gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.gamma` for full documentation
+            - :meth:`numpy.random.RandomState.gamma`
+        """
+        return self.standard_gamma(shape, size) * scale
+
+    def standard_gamma(self, shape, size=None, dtype=numpy.float64, out=None):
+        """Returns an array of samples drawn from a standard gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.standard_gamma` for full documentation
+            - :meth:`numpy.random.RandomState.standard_gamma`
+        """
+        cdef ndarray y
+
+        if shape < 0.0:
+            raise ValueError('shape < 0')
+
+        if out is None or out.dtype.char != 'd':
+            y = ndarray(size if size is not None else (), numpy.float64)
+
+        if out is not None:
+            self._check_output_array(dtype, size, out)
+            if out.dtype.char == 'd':
+                y = out
+
+        if numpy.dtype(dtype).char not in ('f', 'd'):
+            raise TypeError(
+                f'Unsupported dtype {y.dtype.name} for standard_normal')
+
+        y = ndarray(size if size is not None else (), numpy.float64)
+        _launch_dist(self.bit_generator, standard_gamma, y, (shape,))
+        if out is not None and y is not out:
```

Comment:
I think at this point this is safe:
```suggestion
        if y is not out:
```

Reply:
It was not 😂, 
`None` is not `y`, so we try to copy stuff in `None` and the tests failed. I thought so at the beginning too :)

Reply:
Ah ok 😂